statistics-rest/src/main/java/org/uniprot/api/statistics/service/StatisticService.java

```diff
@@ -0,0 +1,85 @@
+        if (categories.isEmpty()) {
+            entries =
+                    statisticsEntryRepository.findAllByReleaseNameAndEntryType(
+                            version, statisticMapper.map(getStatisticType(statisticType)));
+        } else {
+            entries =
+                    statisticsEntryRepository
+                            .findAllByReleaseNameAndEntryTypeAndStatisticsCategoryIn(
+                                    version,
+                                    statisticMapper.map(getStatisticType(statisticType)),
+                                    categories.stream()
+                                            .map(
+                                                    cat ->
+                                                            statisticsCategoryRepository
+                                                                    .findByCategoryIgnoreCase(cat)
+                                                                    .orElseThrow(
+                                                                            () ->
+                                                                                    new IllegalArgumentException(
+                                                                                            String
+                                                                                                    .format(
+                                                                                                            "Invalid Statistic Category: %s",
+                                                                                                            cat))))
+                                            .collect(Collectors.toList()));
+        }
+        return entries.stream()
+                .collect(Collectors.groupingBy(UniprotkbStatisticsEntry::getStatisticsCategory))
+                .entrySet()
+                .stream()
+                .map(entry -> StatisticModuleStatisticCategoryImpl.builder()
+                        .categoryName(entry.getKey().getCategory())
+                        .totalCount(
+                                entry.getValue().stream().mapToLong(UniprotkbStatisticsEntry::getValueCount).sum())
+                        .label(entry.getKey().getLabel())
+                        .searchField(entry.getKey().getSearchField())
+                        .items(
+                                entry.getValue().stream()
+                                        .map(statisticMapper::map)
+                                        .collect(Collectors.toList()))
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private static StatisticModuleStatisticType getStatisticType(String statisticType) {
+        try {
+            return StatisticModuleStatisticType.valueOf(statisticType.toUpperCase());
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
```

Comment:
We should catch specific exception here e.g. IllegalArgumentException. What can throw other exception type?

Reply:
Here it should be either IllegalArgumentException (by passing an invalid string: not a valid enum), or NullPointerException (by passing a null string). In either way IllegalArgumentException looks correct for me,
Please correct me if I'm wrong.